src/x11/screen.c

```diff
@@ -64,6 +64,27 @@ static double get_xft_dpi_value(void)
         return dpi;
 }
 
+static double screen_dpi_get_from_monitor(struct screen_info *scr)
+{
+        return (double)scr->h * 25.4 / (double)scr->mmh;
+}
+
+double screen_dpi_get(struct screen_info *scr)
+{
+        if (  ! settings.force_xinerama
+             && settings.per_monitor_dpi)
+                return screen_dpi_get_from_monitor(scr);
+
+        if (screen_dpi_get_from_xft() > 0)
+                return screen_dpi_get_from_xft();
+
+        /* Calculate the DPI on the overall screen size.
+         * xrandr --dpi <DPI> does only change the overall screen's millimeters,
+         * but not the physical screen's sizes. */
+        return ((((double)DisplayWidth  (xctx.dpy, 0)) * 25.4) /
+                 ((double)DisplayWidthMM(xctx.dpy, 0)));
```

Comment:
What happens with multi-monitor setups here? Perhaps `0` should be replaced with `scr->id`?

Reply:
It's explicitly a `0`. Because we're accessing the (virtual?) x11 screen and not a screen linked to a physical monitor. Run

```
> xrandr -q | head -n1
Screen 0: minimum 8 x 8, current 3840 x 1080, maximum 16384 x 16384
```

And this `0` given there in the code "hardcodes" it to that `Screen 0` mentioned by xrandr. While `scr->id` would reference the `nth` active physical monitor.

It "hardcodes" it to the `Screen 0` in this code. Yes, this code doesn't scale for "Screen 1`, this would be a problem. I don't know how to create multiple xrandr screens either. It covers 99.9% of all users DPI problems when you choose to change the DPI with xrandr.

The definitions of the terms of monitor, display, screen and output are a complete mess. Our "screen" is something completely different to the term screen used in xrandr.

Reply:
But yeah, It warrants a comment. I've added it.

Reply:
`0` might not be portable, but `XDefaultScreen` will always report the correct value. So I changed it with this.